rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java

```diff
@@ -239,6 +241,64 @@ public void savePegoutsWaitingForConfirmations() {
         }
     }
 
+    public List<Coin> getReleaseOutpointsValues(Sha256Hash releaseTxHash) {
+        if (!activations.isActive(RSKIP305)) {
+            throw new IllegalStateException("Can't call this method before RSKIP305 activation.");
+        }
+
+        SortedMap<Sha256Hash, List<Coin>> savedReleasesOutpointsValues = safeGetFromRepository(
+            RELEASES_OUTPOINTS_VALUES,
+            BridgeSerializationUtils::deserializeReleasesOutpointsValues
+        );
+
+        List<Coin> releaseOutpointsValues = savedReleasesOutpointsValues.get(releaseTxHash);
+        if (isNull(releaseOutpointsValues)) {
+            logger.error("Outpoints values for release {} don't exist in storage.", releaseTxHash);
+            throw new IllegalArgumentException("Release tx hash key was not found in releases outpoints values map.");
+        }
+        return releaseOutpointsValues;
+    }
+
+    public void setReleaseOutpointsValues(Sha256Hash releaseTxHash, List<Coin> outpointsValues) {
+        if (!activations.isActive(RSKIP305)) {
+            return;
+        }
+
+        if (releaseTxHash == null || outpointsValues == null) {
+            throw new IllegalArgumentException(
+                String.format("Invalid release outpoints values entry, has hash %s and coins list %s", releaseTxHash, outpointsValues)
+            );
+        }
+
+        if (releasesOutpointsValues == null) {
```

Comment:
I think I understand what @nathanieliov is saying.

We are checking that `if (releasesOutpointsValues == null)` when we haven't potentially get the `releasesOutpointsValues` from the storage. If we don't ensure we get it from the storage and set it in this class state before trying to reassign it with `releasesOutpointsValues = new TreeMap<>();` and then we save this new map, we could override other values that might be present in the storage already.

The following is just an idea, take your time to analyze this case and let us know:

```suggestion
        if (releasesOutpointsValues == null) {
        
             SortedMap<Sha256Hash, List<Coin>> savedReleasesOutpointsValues = safeGetFromRepository(
            RELEASES_OUTPOINTS_VALUES,
            BridgeSerializationUtils::deserializeReleasesOutpointsValues
        );
        
        if (savedReleasesOutpointsValues == null) {
            releasesOutpointsValues = new TreeMap<>();
        } else {
          releasesOutpointsValues = savedReleasesOutpointsValues;
        }
        
   }
```